src/platform/silabs/wifi/icd/WifiSleepManager.cpp

```diff
@@ -34,46 +32,16 @@ CHIP_ERROR WifiSleepManager::Init()
     return CHIP_NO_ERROR;
 }
 
-void WifiSleepManager::HandleCommissioningComplete()
-{
-    VerifyAndTransitionToLowPowerMode();
-}
-
-void WifiSleepManager::HandleInternetConnectivityChange()
-{
-    // TODO: Centralize the buisness logic in the VerifyAndTransitionToLowPowerMode
-    if (!isCommissioningInProgress)
-    {
-        VerifyAndTransitionToLowPowerMode();
-    }
-}
-
-void WifiSleepManager::HandleCommissioningWindowClose() {}
-
-void WifiSleepManager::HandleCommissioningSessionStarted()
-{
-    isCommissioningInProgress = true;
-}
-
-void WifiSleepManager::HandleCommissioningSessionStopped()
-{
-    isCommissioningInProgress = false;
-}
-
 CHIP_ERROR WifiSleepManager::RequestHighPerformance()
 {
     VerifyOrReturnError(mHighPerformanceRequestCounter < std::numeric_limits<uint8_t>::max(), CHIP_ERROR_INTERNAL,
                         ChipLogError(DeviceLayer, "High performance request counter overflow"));
 
-    if (mHighPerformanceRequestCounter == 0)
-    {
-#if SLI_SI917 // 917 SoC & NCP
-        VerifyOrReturnError(wfx_power_save(RSI_ACTIVE, HIGH_PERFORMANCE, 0) == SL_STATUS_OK, CHIP_ERROR_INTERNAL,
-                            ChipLogError(DeviceLayer, "Failed to set Wi-FI configuration to HighPerformance"));
-#endif // SLI_SI917
-    }
-
     mHighPerformanceRequestCounter++;
+
+    // We don't do the mHighPerformanceRequestCounter check here; the check is in TransitionToLowPowerMode function
+    ReturnErrorOnFailure(VerifyAndTransitionToLowPowerMode());
```

Comment:
Yes for now because the calls aren't smart enough to fail if it doesn't go in HighPerformance yet.
They continue to operate will remove the high performance request when they are finished / failed.